kornia/augmentation/augmentation.py

```diff
@@ -884,7 +891,26 @@ def compute_transformation(self, input: torch.Tensor, params: Dict[str, torch.Te
         return F.compute_crop_transformation(input, params, self.flags)
 
     def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
-        return F.apply_crop(input, params, self.flags)
+        if self.flags['mode'] == 'resample':
+            return F.apply_crop(input, params, self.flags)
+        elif self.flags['mode'] == 'slice':
+            import kornia as K
+            coords = params['src'].long()
+            B, C, _, _ = input.shape
+            out = torch.empty(B, C, *self.size, device=input.device, dtype=input.dtype)
+            for i in range(B):
+                x1 = int(coords[i, 0, 0])
+                x2 = int(coords[i, 1, 0]) + 1
+                y1 = int(coords[i, 0, 1])
+                y2 = int(coords[i, 3, 1]) + 1
+                out[i] = K.geometry.transform.resize(
+                    input[i:i+1, :, y1:y2, x1:x2],
+                    self.size,
+                    interpolation='bilinear',
+                    align_corners=True)
+            return out
+        else:
+            raise NotImplementedError(f"Not supported type: {self.flags['mode']}.")
```

Comment:
@shijianjian check if something like this would work.
And for augmentations I would give that one by the default so that gives better performance

Reply:
Add tests. We are fine with it if it passed all the tests.

Reply:
Also, can you provide a benchmark between the two implementations?